packages/ckeditor5-engine/src/conversion/downcasthelpers.js

```diff
@@ -357,6 +358,113 @@ export default class DowncastHelpers extends ConversionHelpers {
 	markerToHighlight( config ) {
 		return this.add( downcastMarkerToHighlight( config ) );
 	}
+
+	/**
+	 * Model marker converter for data downcast.
+	 *
+	 * This conversion creates a representation for model marker boundaries in the view:
+	 *
+	 * * if the marker boundary is at a position where text nodes are allowed, then a view element with specified tag name
+	 * and `name` attribute is added at that position,
+	 * * in other case, a specified attribute is set on an view element that is before/after marker boundary.
+	 *
+	 * Tag and attribute names and values are generated from the marker name:
+	 *
+	 * * templates for attributes are `data-group-start-before="name"`, `data-group-start-after="name"`,
+	 * `data-group-end-before="name"` and `data-group-end-after="name"`,
+	 * * templates for view elements are `<group-start name="name">` and `<group-end name="name">`.
+	 *
+	 * Attributes mark whether given marker start or end boundary is before or after given element.
+	 * Attributes `data-group-start-before` and `data-group-end-after` are favored. Other two are used when former two cannot be used.
+	 *
+	 * Typically, the marker names use `group:uniqueId:otherData` convention. For example: `comment:e34zfk9k2n459df53sjl34:zx32c`.
```

Comment:
I'd move that higher. It sets the context. Before "Tag and attribute names..."

Reply:
I had it there originally :).